contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -166,22 +167,22 @@ pub fn try_refund<S: Storage, A: Api, Q: Querier>(
     let msgs = send_tokens(&deps.api, &env.contract.address, &rcpt, swap.balance)?;
     Ok(HandleResponse {
         messages: msgs,
-        log: vec![log("action", "refund"), log("id", id), log("to", rcpt)],
+        attributes: vec![attr("action", "refund"), attr("id", id), attr("to", rcpt)],
         data: None,
     })
 }
 
-fn parse_hex_32(data: &str) -> StdResult<Vec<u8>> {
+fn parse_hex_32(data: &str) -> Result<Vec<u8>, ContractError> {
     match hex::decode(data) {
         Ok(bin) => {
             if bin.len() == 32 {
                 Ok(bin)
             } else {
-                Err(StdError::generic_err("Hash must be 64 characters"))
+                Err(ContractError::Invalid("hash: must be 64 characters".into()))
```

Comment:
For example, here we could use

`ContractError::InvalidHashLength(data.len())` and add  `"hash must be 64 characters, you provided {0}"` inside the `#[error]` text